skiko/src/iosMain/kotlin/org/jetbrains/skiko/redrawer/MetalRedrawer.ios.kt

```diff
@@ -82,14 +126,44 @@ internal class MetalRedrawer(
 
     override fun needRedraw() {
         check(!isDisposed) { "MetalRedrawer is disposed" }
-        caDisplayLink.setPaused(false)
+
+        drawImmediatelyIfPossible()
+
+        if (drawSchedulingState.hasDrawScheduledForNextFrame) {
+            caDisplayLink.setPaused(false)
+        }
     }
 
     override fun redrawImmediately() {
         check(!isDisposed) { "MetalRedrawer is disposed" }
         draw()
     }
 
+    /**
+     * Dispatch redraw immediately during current frame if possible
+     */
+    private fun drawImmediatelyIfPossible() {
+        when (drawSchedulingState) {
+            DrawSchedulingState.AVAILABLE_ON_NEXT_FRAME -> {
+                drawSchedulingState = DrawSchedulingState.SCHEDULED_ON_NEXT_FRAME
+            }
+
+            DrawSchedulingState.AVAILABLE_ON_CURRENT_FRAME -> {
+                drawIfLayerIsShowing()
+
+                drawSchedulingState = DrawSchedulingState.AVAILABLE_ON_NEXT_FRAME
+            }
+
+            DrawSchedulingState.SCHEDULED_ON_NEXT_FRAME -> Unit /** already scheduled, do nothing */
```

Comment:
Remove Unit?

Reply:
What's an idiomatic way to do nothing on empty `when` cases in Kotlin?

Reply:
Done.

Reply:
I think there is no idiomatic way. We need to agree about this within our team. I would use "{}" instead of Unit. @igordmn?

Reply:
Done.

Reply:
I would also use "//" styled comment within {}

Reply:
I guess `Unit` is OK, but it can be replaced with `{}`.

Regarding having this case at all, it's better to keep it explicit instead of just `else`

Reply:
Agree with @MatkovIvan about being explicit on all enum cases.

Will replace /** */ for // inside function scopes.

Reply:
BTW, "/**" are used only for comments that are going to be reflected in public documentation. Two stars mean that it is a public comment.

Reply:
Aha, got it! Do we have some linting tooling, I'm not aware of?

Reply:
I believe that IDE's are configured for code styles (so when you do format code with IDE, it adjusts for code styles) but it does not catch all cases.